flocx_market/tests/commit/test_commit_message.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=missing-docstring,redefined-outer-name,no-self-use,invalid-name
+
+import os
+import re
+import subprocess
+
+import git
+import pytest
+
+RE_PARSE_NAME = re.compile(r'(?P<name>.*) <(?P<email>[^>]+)> '
+                           r'(?P<timestamp>\d+) (?P<tz>[-\d]+)')
+
+
+@pytest.fixture
+def commit_id():
+    '''A dummy fixture that will be parametrized by the pytest_generate_tests
+    function.'''
+
+
+@pytest.fixture
+def repo():
+    '''Return a git.Repo object for the current directory'''
+    return git.Repo('.')
+
+
+@pytest.fixture
+def commit(repo, commit_id):
+    '''Convert the parametrized commit_id into a Commit object'''
+    return repo.commit(commit_id)
+
+
+@pytest.fixture
+def max_subject_length():
+    return int(os.environ.get('COMMIT_MAX_SUBJECT_LENGTH', 60))
+
+
+def get_commits_from_pr():
+    commit_range = os.environ.get('TRAVIS_COMMIT_RANGE')
+    out = subprocess.check_output(['git', 'rev-list', commit_range])
+    out = out.decode('utf-8')
+    commits = out.splitlines()
+
+    return commits
+
+
+def pytest_generate_tests(metafunc):
+    '''Get list of commits involved in current pull request'''
+
+    if os.environ.get('TRAVIS_PULL_REQUEST', 'false') != 'false':
+        commits = get_commits_from_pr()
+    elif 'TRAVIS_COMMIT' in os.environ:
+        commits = [os.environ.get('TRAVIS_COMMIT')]
+    else:
+        commits = ['HEAD']
+
+    metafunc.parametrize('commit_id', commits)
+
+
+def test_commit_subject_and_body(commit):
+    '''Verify that a commit has a subject and body separated by a blank line'''
+    message = commit.message.splitlines()
+    assert len(message) > 2, (
+        'Commit message must include both subject and body')
+
+    assert not message[1], (
+        'Commit message must have a blank line after the subject')
+
+
+def test_commit_subject_no_period(commit):
```

Comment:
I don't fully understand the need for this check, but I'm not strongly against it if there's some convention I'm unaware of... ?

Reply:
It's a common convention for commit subjects.  E.g. [in openstack](https://wiki.openstack.org/wiki/GitCommitMessages#Summary_of_Git_commit_message_structure).